src/ucp/core/ucp_mm.h

```diff
@@ -59,8 +59,9 @@ void ucp_rkey_resolve_inner(ucp_rkey_h rkey, ucp_ep_h ep);
 
 #define UCP_RKEY_RESOLVE(_rkey, _ep, _op_type) \
     ({ \
-        ucs_status_t status = UCS_OK; \
-        if (ucs_unlikely((_ep)->cfg_index != (_rkey)->cache.ep_cfg_index)) { \
+        ucs_status_t status = ucp_ep_config(_ep)->status; \
```

Comment:
why? the new cfg_index should point to a configuration with the new set of lanes (where the failed lane at index 0 and rest are removed), so rkey lookup should work. then it would call the put/get/amo operations of failed lane, and return the status..